etc/cvmfs/domain.d/computecanada.ca.conf

```diff
@@ -0,0 +1,15 @@
+# These are here so cvmfs will notice them if common.conf sets them
+CVMFS_HTTP_PROXY="$CVMFS_HTTP_PROXY"
+CVMFS_FALLBACK_PROXY="$CVMFS_FALLBACK_PROXY"
+. ../common.conf
+
+if [ "$CVMFS_USE_CDN" = "yes" ] || [ "$CVMFS_HTTP_PROXY" = "DIRECT" ] || [ "$CVMFS_HTTP_PROXY" = "auto;DIRECT" ]; then
+  CVMFS_SERVER_URL="http://cvmfs-s1.computecanada.net/cvmfs/@fqrn@"
+  CVMFS_USE_GEOAPI=no
+else
+  CVMFS_SERVER_URL="http://cvmfs-s1-arbutus.computecanada.ca:8000/cvmfs/@fqrn@;http://cvmfs-s1-beluga.computecanada.ca:8000/cvmfs/@fqrn@;http://cvmfs-s1-graham.computecanada.ca:8000/cvmfs/@fqrn@;http://cvmfs-s1.computecanada.net/cvmfs/@fqrn@"
```

Comment:
I see this was copied from the `default` config-repo, but I think it may be a mistake to include the cloudflare alias cvmfs-s1.computecanada.net in this setup because it will probably confuse the Geo API, and depending on the luck of where you query it from, it may get sorted to be before some of the direct stratum 1s.

@jblomer @rptaylor do you have any comment?  Maybe I'm wrong, maybe the cloudflare alias always ends up last.  I just tried some tests and that appears to be the case.  If so, I think it is accidental.

Reply:
cvmfs-s1.computecanada.net  is a load balancer of all the other stratum 1 servers listed here (and one other also). It uses [dynamic steering](https://developers.cloudflare.com/load-balancing/understand-basics/traffic-steering/steering-policies/dynamic-steering/) to determine the closest origin. If an origin server is unavailable, the [zero-downtime](https://blog.cloudflare.com/new-tools-to-monitor-your-server-and-avoid-downtime/) feature dynamically redirects a client to another origin within the same HTTP request to avoid serving an error to clients.
 
 I don't think there is a great solution to reconciling the different ways that CVMFS and Cloudflare do geo-location, but the impetus for including it in CVMFS_SERVER_URL (in the case when proxies are used) is mainly for DDoS resilience. When I did some performance microbenchmarks comparing the Cloudflare stratum server to other servers, sometimes it could be faster, sometimes slower, depending on the client location and access patterns. Overall I think the case for including it outweighs the possibility that the connection might (or actually might not) be slightly sub-optimal, and anyway for use outside Canada I think the potential upside should be higher.

Reply:
My main motivation to avoid going through Cloudflare when using a local proxy is that there is no logging for debugging problems and watching for (usually accidental) abuse.  Especially if the local proxy goes down and all the traffic goes directly to the stratum 1, I want to be alerted of that as soon as possible to make sure it gets repaired.  Do you have that issue?  We additionally have monitored fallback proxies and configure clients to use those and never to go directly to servers when there is a local proxy.

We have so far not seen any intentional denial of service attack on stratum 1s.

Reply:
In this case OSG sites would not have access to the logs anyway (whether for stratum servers or Cloudflare), since they are operated by CC.  Shielding against abuse (whether usually accidental or otherwise) and load spikes from proxy failures is another reason to include Cloudflare, isn't it?

We do have a log analytics system (filebeat and Kibana I believe) that could be used to look for anomalous connections but in practice it hasn't been necessary and we haven't had any of those kinds of operational issues.

The other main impetus for using Cloudflare was to extend our reach and presence to anywhere in the world the repo is used, whereas our stratum servers are only located in Canada. It sounds like operating an OSG stratum server that replicates the CC repo(s) would provide what you are looking for (regional network connectivity and access to log files for debugging), which is fine if you want to do that, although using Cloudflare seems simpler and easier to me.

Reply:
There might be good reasons for replicating the repos further around the world, but I wasn't thinking about us needing the logs from your servers, I was thinking that you might have good reason to need them.  We experience site proxy failures every week in the WLCG and are interested in not only protecting our servers but also protecting the WAN usage of our sites (and also partly wanting to avoid causing enough hits onto Cloudflare that they will require us to move up from the free tier).   We get this already by our monitoring of the fallback proxies and our stratum 1s.

In any case it looks to me like having the Cloudflare entry here is unlikely to have much effect since it is always geosorted to the end, so it's fine.